drop-transfer/src/ws/client/mod.rs

```diff
@@ -224,7 +233,7 @@ async fn establish_ws_conn(
             Ok(_) => break ver,
             Err(RequestError::General(err)) => {
                 info!(logger, "Error while making the HTTP request: {err:?}");
-                return WsConnection::Recoverable;
+                return WsConnection::Recoverable(crate::Error::UnrecognizablePeerRespone);
```

Comment:
`RequestError` has `UnexpectedResponse` which by the name of it should map to `UnrecognizablePeerResponse` instead. IMO the `General` case is for the other occasions

Reply:
Yeah, true. Do you suggest creating another error type for this one?

Reply:
By looking into the code it doesn't make sense to change this. The only thing I see fit for a change is the name `crate::Error::UnrecognizablePeerRespone`,
The `RequestError::UnexpectedResponse` is handled cleanly in this loop, no need to report it

Reply:
I think there's confusion between `UnexpectedResponse` and `UnrecognizablePeerResponse`. Do you suggest to leave it as it is?

Reply:
Yeah, let me think about the naming more

Reply:
I've changed the name to `ConnectionClosedByPeer`